src/main/java/de/tum/in/www1/artemis/service/metis/conversation/OneToOneChatService.java

```diff
@@ -0,0 +1,83 @@
+package de.tum.in.www1.artemis.service.metis.conversation;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.metis.ConversationParticipant;
+import de.tum.in.www1.artemis.domain.metis.conversation.OneToOneChat;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationParticipantRepository;
+import de.tum.in.www1.artemis.repository.metis.conversation.OneToOneChatRepository;
+
+@Service
+public class OneToOneChatService {
+
+    public static final String ONE_TO_ONE_CHAT_ENTITY_NAME = "messages.oneToOneChat";
+
+    private final ConversationParticipantRepository conversationParticipantRepository;
+
+    private final OneToOneChatRepository oneToOneChatRepository;
+
+    private final UserRepository userRepository;
+
+    public OneToOneChatService(ConversationParticipantRepository conversationParticipantRepository, OneToOneChatRepository oneToOneChatRepository, UserRepository userRepository) {
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.oneToOneChatRepository = oneToOneChatRepository;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * Tries to find a OneToOneChat between two users
+     *
+     * @param courseId the course the OneToOneChat is in
+     * @param userAId  the id of the first user
+     * @param userBId  the id of the second user
+     * @return the OneToOneChat if it exists, otherwise empty optional
+     */
+    public Optional<OneToOneChat> findOneToOneChatWithSameMembers(Long courseId, Long userAId, Long userBId) {
+        return oneToOneChatRepository.findBetweenUsersWithParticipantsAndUserGroups(courseId, userAId, userBId);
```

Comment:
Same as above.

Reply:
Done in 8eae2ed6fb3b63c7c7595178f24dfb387a25e4b4